### ISSUE TITLE
Mount configuration not working with third subfolder

### ISSUE THREAD

**Issue by totkeks** [gohugoio/hugo#6346]
I'm trying to mount some additional files in the static target, so that they are published with the Hugo build, even though they come from other folders (javascript packages).

This works and copies the icons folder in the `assets/icons` folder in the output.
```
  - source: src/hugo/static
    target: static
  - source: src/core/assets/icons
    target: static/assets/icons
```

But when I change it to this, then I get nothing of the icons in the output.
```
  - source: src/hugo/static
    target: static
  - source: src/core/assets/icons
    target: static/assets/prefix/icons
```

Is there a limit on the number of subfolders that can be specified for a mount point?

**Comment by bep** [comment]
>Is there a limit on the number of subfolders that can be specified for a mount point?

No, but for the mounts that where we walk the directory tree (e.g. static), you currently cannot have "holes" in the target directory tree as it isn't reachable by traditional directory traversal, so to speak. This is fixable, but I also suspect that you could workaround it by somehow creating some dummy directories. Not sure.

**Comment by bep** [comment]
That said, depending on your use case, you can mount these folders into `/assets` and load them with `resources.Get`, `resources.Match` etc. from your templates. This should work fine even for deeply nested structures.

**Comment by totkeks** [comment]
> > Is there a limit on the number of subfolders that can be specified for a mount point?
> 
> No, but for the mounts that where we walk the directory tree (e.g. static), you currently cannot have "holes" in the target directory tree as it isn't reachable by traditional directory traversal, so to speak. This is fixable, but I also suspect that you could workaround it by somehow creating some dummy directories. Not sure.

Thanks for the quick reply. Indeed, your suggested workaround is working fine. Hugo is copying the files now on build.

**Comment by XhmikosR** [comment]
I believe this issue would help us with https://github.com/twbs/bootstrap/pull/29283

**Comment by jmooring** [comment]
I am unable to reproduce the reported behavior with v0.83.1. 

This test mounts multiple directories under `assets` and `static`, with and without holes in the target path.

```bash
git clone --single-branch -b hugo-github-issue-6346 https://github.com/jmooring/hugo-testing hugo-github-issue-6346
cd hugo-github-issue-6346
hugo server
```

Can we close this issue?

**Comment by XhmikosR** [comment]
It's been a while since I last tried to use this, so it might have been fixed. Just a note that I was testing mostly on Windows, in case that makes any difference.
